### ISSUE TITLE
when smartparens mode enabled, type ' always insert \'\' 

### ISSUE THREAD

**Issue by MaYue-cs** [Fuco1/smartparens#815]
## Expected behavior
type ' key
will insert ''

## Actual behavior
type ' key
\'\'

## Steps to reproduce the problem

always have this problem.
don't know how to configure

## Backtraces if necessary (`M-x toggle-debug-on-error`)
no error

## Environment & version information

**In recent enough `smartparens` you can call `M-x sp-describe-system` to generate this report**.  Please fill manually what we could not detect automatically.  Edit the output as you see fit to protect your privacy.

- `smartparens` version:
- Active major-mode:
- Emacs version (`M-x emacs-version`):
- Spacemacs/Evil/Other starterkit (specify which)/Vanilla:
- OS:

**Comment by Fuco1** [comment]
Can you please fill the `Environment & version information` section?  Hard to debug with pretty much no information.

**Comment by CSRaghunandan** [comment]
I'm also facing the same problem.
Smartparens version: 20171126.1241
I'm running emacs compiled for master branch on Mac OS Sierra.
I'm not using any startkits. Just good old vanilla emacs with my own customization.

**Comment by Fuco1** [comment]
@CSRaghunandan Does this happen everywhere or only in specific major modes?  What is the emacs version?

**Comment by CSRaghunandan** [comment]
@Fuco1 This seems to happen only in c-based modes for me, checked `rust`, `python`, and `js2-mode` and all of them are fine.

emacs version: GNU Emacs 27.0.50 (build 1, x86_64-apple-darwin16.7.0, NS appkit-1504.83 Version 10.12.6 (Build 16G1036)) of 2017-11-27

**Comment by CSRaghunandan** [comment]
@Fuco1 Yeah, I can understand.

Instead of moving to latest stable release, I would rather make an exception for closing of `'` in c and cpp modes since those are the only ones affected. I see that there is a `sp-local-pair` variable, but how do I use it?

**Comment by Fuco1** [comment]
Right.  You can do `(sp-local-pair 'c-mode "'" nil :actions nil)`, that should do the trick.
